### DISCUSSION TITLE
Need to know how depth works in scrapy please help

### DISCUSSION THREAD

**Discussion by AhangarAamir** [scrapy/scrapy#6580] [Q&A] [answered]
Can someone tell me how depth works for e.g if i set DEPTH_LIMIT= 1 or 2 or 3 so on what will happen

**Comment by Gallaecio** [comment, marked as answer]
Let’s say you have a website with the following pages:

```
/a: Links to /b and /d
/b: Links to /c
/c: Links to /d and /e
/d (no links)
/e (no links)
/f (no links)
```

And your spider has  `/a` and `/f` in `start_urls`.

With `DEPTH_LIMIT=1`, only 1 link is followed from the first level of crawl. `/a` and `/f` are the first level of crawl, and since `/a` links to `/b` and `/d`, both of those are parsed, but links extracted from `/b` are ignored due to the depth limit, so `/c` is not reached. Neither is `/e`, since nothing parsed links to it.

With `DEPTH_LIMIT=2`, links from `/b` are also followed, so `/c` is also parsed, but the link to `/e` from `/c` is not followed.

With `DEPTH_LIMIT=3`, links from `/c` are also followed, so `/e` is also parsed.

**Reply by AhangarAamir** [reply]
@Gallaecio Thanks for your reply
I want to crwal root urls for examples I have urls list as [abc.com, def.com, fgh.com] what will be the DEPTH_LIMIT

**Reply by Gallaecio** [reply]
If you do not want to crawl any of their links beyond the home page, you should instead not yield any request from your spider callback(s) to begin with. Either that, or set `DEPTH_LIMIT=1` and implement the `start_requests()` method so that your start requests will include `{"depth": 1}` in `Request.meta`.
